src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -508,9 +534,14 @@ const OfflineTokenRow = ({ onValueChanged, validationFailed }) => {
             setValidationState(validationStates.INPROGRESS);
             getAccessToken(offlineToken)
                     .then(out => {
+                        console.log(out);
                         const accessToken = out.trim();
                         onValueChanged("accessToken", accessToken);
                         setValidationState(validationStates.SUCCESS);
+
+                        // TODO encrypt offline token
+                        return saveOfflineToken(offlineToken)
+                                .catch(ex => console.warn(`Could not save offline token to a storage: "${JSON.stringify(ex)}"`));
                     })
                     .catch(ex => {
                         console.error(`Offline token validation failed: "${JSON.stringify(ex)}"`);
```

Comment:
If the token is invalid, it should be cleared from localStorage. I suppose the most common reason is that is lifetime is over?

In general, we need to take care to clean up the tokens somehow. Doing that here should be enough.